### PR TITLE
Add README for s2i tasks and refactoring

### PR DESCRIPTION
**Pull request by piyush-garg**
This will add readme for the s2i tasks
available.

Also refactored a bit to separate
dotnet1 and dotnet2 tasks

### CODE REVIEW

File: s2i-dotnet/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Source-to-Image
+
+[Source-to-Image (S2I)](https://github.com/openshift/source-to-image)
+is a toolkit and workflow for building reproducible container images
```
Comment (Preeticp) [inline s2i-dotnet/README.md]: a workflow

File: s2i-dotnet/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+from source code. S2I produces images by injecting source code into a
+base S2I container image and letting the container prepare that source
+code for execution. The base S2I container images contains the
```
Comment (Preeticp) [inline s2i-dotnet/README.md]: contains/comprise

File: s2i-dotnet/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+```
+kubectl apply -f https://raw.githubusercontent.com/openshift/pipelines-catalog/master/s2i-dotnet/s2i-dotnet-task.yaml
+```
+
+## Inputs
+
+### Parameters
+
+* **PATH_CONTEXT**: Source path from where s2i command need to be run
```
Comment (Preeticp) [inline s2i-dotnet/README.md]: the s2i command needs to...

File: s2i-dotnet/README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+### Resources
+
+* **source**: A `git`-type `PipelineResource` specifying the location of the
+  source to build.
+
+## Outputs
+
+### Resources
+
+* **image**: An `image`-type `PipelineResource` specify the image that should
```
Comment (Preeticp) [inline s2i-dotnet/README.md]: 'specifying' to keep it consistent with the explanation on line 29

File: s2i-dotnet/README.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  be built.
+
+## ServiceAccount
+
+S2I builds an image and pushes it to the destination registry which is
+defined as a parameter. In order to properly authenticate to the
+remote container registry, it needs to have the proper
+credentials. The credentials can be provided through a service
```
Comment (Preeticp) [inline s2i-dotnet/README.md]: Suggest: The image needs proper credentials to be authenticated by the remote container registry. These credentials can be...

File: s2i-dotnet/README.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+account. See
+[Authentication](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#basic-authentication-docker)
+for further details.
+
+If you are running on OpenShift, you also need to allow the service
```
Comment (Preeticp) [inline s2i-dotnet/README.md]: If you run on...privileged containers. Due to security considerations OpenShift does not allow containers to run as privileged containers by default.

File: s2i-dotnet/README.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+```
+oc create serviceaccount pipeline
+oc adm policy add-scc-to-user privileged -z pipeline
+oc adm policy add-role-to-user edit -z pipeline
+```
+
+## Usage
+
+This TaskRun runs the Dotnet Task to fetch a Git repo, and build and push a
+container image using s2i and a dotnet builder image.
```
Comment (Preeticp) [inline s2i-dotnet/README.md]: Git repository and builds and pushes a...

File: s2i-dotnet/README.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
+# Source-to-Image
+
+[Source-to-Image (S2I)](https://github.com/openshift/source-to-image)
+is a toolkit and workflow for building reproducible container images
+from source code. S2I produces images by injecting source code into a
+base S2I container image and letting the container prepare that source
```
Comment (Preeticp) [inline s2i-dotnet/README.md]: Suggest: S2I injects code into a base S2I container image and lets the container prepare that source code for execution to produce images.
Also please use either s2i or S2I consistently through out the doc.

File: s2i-dotnet/README.md (hunk 1 above)
Comment (Preeticp) [reply s2i-dotnet/README.md]: Is 'and' correct here? Should it have been 'or' instead? As in - a toolkit or a workflow

**Comment by Preeticp** [comment]
LGTM

File: s2i-dotnet/README.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
+# Source-to-Image
```
Comment (vdemeester) [inline s2i-dotnet/README.md]: Shouldn't this be `Dotnet Source-to-Image`, with a small change on the paragraph below on the specifics of this task focused on `dotnet` ?

**Review by vdemeester — COMMENTED** [review]
@piyush-garg one comment :angel:

**Review by siamaksade — APPROVED** [review]

File: s2i-dotnet/README.md (hunk 9 above)
Comment (siamaksade) [reply s2i-dotnet/README.md]: Yes. that would be much better

File: s2i-dotnet1/README.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
+# Dotnet1 Source-to-Image
```
Comment (siamaksade) [inline s2i-dotnet1/README.md]: .NET Core 1 Source-to-Image

File: s2i-dotnet1/README.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+This task can be used for building `.NET1` Core apps as reproducible Docker
```
Comment (siamaksade) [inline s2i-dotnet1/README.md]: `.NET Core 1`

File: s2i-dotnet1/README.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
+# Dotnet1 Source-to-Image
+
+This task can be used for building `.NET1` Core apps as reproducible Docker 
+images using Source-to-Image. [Source-to-Image (S2I)](https://github.com/openshift/source-to-image)
+is a toolkit and a workflow for building reproducible container images
+from source code. This tasks uses the s2i-dotnet image build from [redhat-developer/s2i-dotnetcore](https://github.com/redhat-developer/s2i-dotnetcore).
+
+.NET1 Core versions currently provided are:
```
Comment (siamaksade) [inline s2i-dotnet1/README.md]: .NET Core versions...

File: s2i-dotnet2/README.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
+# Dotnet2 Source-to-Image
```
Comment (siamaksade) [inline s2i-dotnet2/README.md]: .NET Core 2 Source-to-Image

File: s2i-dotnet2/README.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+
+This task can be used for building `.NET2` Core apps as reproducible Docker
```
Comment (siamaksade) [inline s2i-dotnet2/README.md]: .NET Core 2

File: s2i-dotnet2/README.md [hunk 15]
```diff
@@ -0,0 +1,96 @@
+# Dotnet2 Source-to-Image
+
+This task can be used for building `.NET2` Core apps as reproducible Docker 
+images using Source-to-Image. [Source-to-Image (S2I)](https://github.com/openshift/source-to-image)
+is a toolkit and a workflow for building reproducible container images
+from source code. This tasks uses the s2i-dotnet image build from [redhat-developer/s2i-dotnetcore](https://github.com/redhat-developer/s2i-dotnetcore).
+
+.NET2 Core versions currently provided are:
```
Comment (siamaksade) [inline s2i-dotnet2/README.md]: .NET Core versions...

File: s2i-dotnet1/README.md [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+
+- 1.0 (RHEL 7, CentOS 7)
+- 1.1 (RHEL 7)
+
+## Installing the Dotnet1 Task
```
Comment (siamaksade) [inline s2i-dotnet1/README.md]: ... .NET Core 1 Task

File: s2i-dotnet1/s2i-dotnet1-task.yaml [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+  name: s2i-dotnet1
+spec:
+  inputs:
+    resources:
+      - name: source
+        type: git
+    params:
+      - name: VERSION
+        description: The minor version of the dotnet1
+        default: '1'
```
Comment (siamaksade) [inline s2i-dotnet1/s2i-dotnet1-task.yaml:13]: Param should be the full version and default to: 1.1

File: s2i-dotnet2/s2i-dotnet2-task.yaml [hunk 18]
```diff
@@ -1,13 +1,16 @@
...
+  name: s2i-dotnet2
 spec:
   inputs:
     resources:
       - name: source
         type: git
     params:
+      - name: VERSION
+        description: The minor version of the dotnet2
+        default: '2'
```
Comment (siamaksade) [inline s2i-dotnet2/s2i-dotnet2-task.yaml:13]: Param should be the full version and default to 2.2

File: s2i-java11/README.md [hunk 19]
```diff
@@ -0,0 +1,89 @@
+# Java11 Source-to-Image
```
Comment (siamaksade) [inline s2i-java11/README.md]: Java 11  (space added)

File: s2i-java8/README.md [hunk 20]
```diff
@@ -0,0 +1,91 @@
+# Java8 Source-to-Image
```
Comment (siamaksade) [inline s2i-java8/README.md]: Java 8

**Review by siamaksade — APPROVED** [review]

**Pull request merged**